walletnode/services/artworkregister/task.go

```diff
@@ -198,6 +218,40 @@ func (task *Task) run(ctx context.Context) error {
 	return nil
 }
 
+// generateDDAndFingerprintsIDs generates redundant IDs and assigns to task.redundantIDs
+func (task *Task) generateDDAndFingerprintsIDs() error {
+	ddDataJSON, err := json.Marshal(task.fingerprintAndScores)
+	if err != nil {
+		return errors.Errorf("failed to marshal dd-data: %w", err)
+	}
+
+	ddEncoded := utils.B64Encode(ddDataJSON)
+
+	var buffer bytes.Buffer
+	buffer.Write(ddEncoded)
+	buffer.WriteByte(pastel.SeparatorByte)
+	buffer.Write(task.signatures[0])
+	buffer.WriteByte(pastel.SeparatorByte)
+	buffer.Write(task.signatures[1])
+	buffer.WriteByte(pastel.SeparatorByte)
+	buffer.Write(task.signatures[2])
+	ddFpFile := buffer.Bytes()
+
+	task.ddAndFingerprintsIc = rand.Uint32()
+	task.ddAndFingerprintsIDs, _, err = pastel.GetIDFiles(ddFpFile, task.ddAndFingerprintsIc, task.config.DDAndFingerprintsMax)
+	if err != nil {
+		return fmt.Errorf("get ID Files: %w", err)
+	}
+
+	comp, err := zstd.CompressLevel(nil, ddFpFile, 22)
+	if err != nil {
+		return errors.Errorf("compress: %w", err)
+	}
+	task.ddAndFpFile = utils.B64Encode(comp)
```

Comment:
while we need to encode compress to base64?

Reply:
@nguyen202 please read step 11A

Reply:
Thanks.